packages/opentelemetry-exporter-collector/src/types.ts

```diff
@@ -14,485 +14,156 @@
-   */
-  functionName?: TruncatableString;
-  /**
-   * An un-mangled function name, if `function_name` is
-   * [mangled](http://www.avabodh.com/cxxin/namemangling.html). The name can be
-   * fully qualified.
-   */
-  originalFunctionName?: TruncatableString;
-  /**
-   * The name of the source file where the function call appears.
-   */
-  fileName?: TruncatableString;
-  /**
-   * The line number in `file_name` where the function call appears.
-   */
-  lineNumber?: string;
-  /**
-   * The column number where the function call appears, if available. This is
-   * important in JavaScript because of its anonymous functions.
-   */
-  columnNumber?: string;
-  /**
-   * The binary module from where the code was loaded.
-   */
-  loadModule?: Module;
-  /**
-   * The version of the deployed source code.
-   */
-  sourceVersion?: TruncatableString;
-}
-
-/**
- * A collection of stack frames, which can be truncated.
- */
-export interface StackFrames {
-  /**
-   * Stack frames in this call stack.
-   */
-  frame?: StackFrame[];
-  /**
-   * The number of stack frames that were dropped because there were too many
-   * stack frames. If this value is 0, then no stack frames were dropped.
-   */
-  droppedFramesCount?: number;
-}
-
-/**
- * The call stack which originated this span.
- */
-export interface StackTrace {
-  /**
-   * Stack frames in this stack trace.
-   */
-  stackFrames?: StackFrames;
-  /**
-   * The hash ID is used to conserve network bandwidth for duplicate stack
-   * traces within a single trace.  Often multiple spans will have identical
-   * stack traces. The first occurrence of a stack trace should contain both
-   * `stack_frames` and a value in `stack_trace_hash_id`.  Subsequent spans
-   * within the same request can refer to that stack trace by setting only
-   * `stack_trace_hash_id`.
-   */
-  stackTraceHashId?: string;
-}
-
-/**
- * A time-stamped annotation or message event in the Span.
- */
-export interface TimeEvent {
-  /**
-   * The time the event occurred.
-   */
-  time?: string;
-  /**
-   * A text annotation with a set of attributes.
-   */
-  annotation?: Annotation;
-  /**
-   * An event describing a message sent/received between Spans.
-   */
-  messageEvent?: MessageEvent;
-}
-
-/**
- * A collection of `TimeEvent`s. A `TimeEvent` is a time-stamped annotation on
- * the span, consisting of either user-supplied key-value pairs, or details of a
- * message sent/received between Spans.
- */
-export interface TimeEvents {
-  /**
-   * A collection of `TimeEvent`s.
-   */
-  timeEvent?: TimeEvent[];
-  /**
-   * The number of dropped annotations in all the included time events. If the
-   * value is 0, then no annotations were dropped.
-   */
-  droppedAnnotationsCount?: number;
-  /**
-   * The number of dropped message events in all the included time events. If
-   * the value is 0, then no message events were dropped.
-   */
-  droppedMessageEventsCount?: number;
+export namespace opentelemetryProto {
```

Comment:
Just for interest: Are these types here handwritten or generated from .proto files? I haven't seen a script to generate them.

Reply:
handwritten